src/aosm/azext_aosm/build_processors/helm_chart_processor.py

```diff
@@ -0,0 +1,384 @@
+                    image_name,
+                    image_tag,
+                    self.name,
+                )
+                images.append((image_name, image_tag))
+            else:
+                logger.warning(
+                    "Could not parse image name and tag in line %s in Helm chart %s",
+                    line,
+                    self.name,
+                )
+
+        return images
+
+    def _find_image_lines(self, chart: HelmChartInput, image_lines: Set[str]) -> None:
+        """
+        Finds the lines containing image references in the given Helm chart and its dependencies.
+
+        :param image_lines: A set of image lines found so far.
+        :type image_lines: Set[str]
+        """
+        logger.debug("Finding image lines in Helm chart %s", chart.artifact_name)
+        # Find the image lines in the current chart
+        for template in chart.get_templates():
+            for line in template.data:
+                if "image:" in line:
+                    logger.debug(
+                        "Found image line %s in Helm chart %s",
+                        line,
+                        chart.artifact_name,
+                    )
+                    image_lines.add(line.replace("image:", "").strip())
+
+        # Recursively search the dependency charts for image lines
+        for dep in self.input_artifact.get_dependencies():
+            self._find_image_lines(dep, image_lines)
+
+    def _generate_artifact_profile(self) -> AzureArcKubernetesArtifactProfile:
+        """
+        Generates an Azure Arc Kubernetes artifact profile for the given artifact store and Helm chart.
+
+        :return: The generated Azure Arc Kubernetes artifact profile.
+        :rtype: AzureArcKubernetesArtifactProfile
+        """
+        logger.debug("Generating artifact profile for Helm chart input.")
+        image_pull_secrets_values_paths: Set[str] = set()
+        self._find_image_pull_secrets_values_paths(
+            self.input_artifact, image_pull_secrets_values_paths
+        )
+
+        registry_values_paths = self._find_registry_values_paths()
+
+        helm_artifact_profile = HelmArtifactProfile(
+            helm_package_name=self.input_artifact.artifact_name,
+            helm_package_version_range=self.input_artifact.artifact_version,
+            registry_values_paths=list(registry_values_paths),
+            image_pull_secrets_values_paths=list(image_pull_secrets_values_paths),
+        )
+
+        # We set the artifact store ID as an empty string as the builder will
+        # set the artifact store ID when it writes the bicep template for the NFD.
+        return AzureArcKubernetesArtifactProfile(
+            artifact_store=ReferencedResource(id=""),
+            helm_artifact_profile=helm_artifact_profile,
+        )
+
+    def _find_image_pull_secrets_values_paths(
+        self, chart: HelmChartInput, matches: Set[str]
+    ) -> None:
```

Comment:
This is another slightly unusual method that doesn't return anything, nor update an instance variable, so presumably directly updates a variable passed by reference. 

If so, I think that should be in the doc string, and possibly in the places that call this method too, as it's really unusual in our code.

Reply:
I'm not convinced this method does anything useful!

It doesn't return anything.
It doesn't set any parameters on the instance (the only instance of `self` is the recursive call to `self._find_image_pull_secrets_values_paths()` .

In the end, we just delete imagePullSecrets (because NFM supplies them), and that removal method doesn't worry at all about all this recursive searching. I think this can all be deleted.